File: backend/internal/model/sede.go
package model

import (
	"database/sql"
	
)

type Sede struct {
	TenantModel

	Nombre     string     `json:"nombre" db:"nombre"`
	Slug       string     `json:"slug" db:"slug"`
	CodigoSede NullString `json:"codigo_sede" db:"codigo_sede"`

	// Dirección
	Direccion    string     `json:"direccion" db:"direccion"`
	Departamento NullString `json:"departamento" db:"departamento"`
	Provincia    NullString `json:"provincia" db:"provincia"`
	Distrito     NullString `json:"distrito" db:"distrito"`
	Referencia   NullString `json:"referencia" db:"referencia"`

	// Contacto
	Telefono          NullString `json:"telefono" db:"telefono"`
	Email             NullString `json:"email" db:"email"`
	ResponsableNombre NullString `json:"responsable_nombre" db:"responsable_nombre"`
	ResponsableCargo  NullString `json:"responsable_cargo" db:"responsable_cargo"`

	// Horario (JSONB) - usar sql.NullString para manejar NULL
HorarioAtencion sql.NullString `json:"horario_atencion" db:"horario_atencion"`

	// Geolocalización
	Latitud  NullFloat64 `json:"latitud" db:"latitud"`
	Longitud NullFloat64 `json:"longitud" db:"longitud"`

	// Estado
	Activo      bool `json:"activo" db:"activo"`
	EsPrincipal bool `json:"es_principal" db:"es_principal"`

	Timestamps
}

File: frontend/src/tipos/sede.ts
import type { Nullable } from './api';

export interface Sede {
  id: string;
  tenant_id: string;
  nombre: string;
  slug: string;
  codigo_sede: Nullable<string>;
  direccion: string;
  departamento: Nullable<string>;
  provincia: Nullable<string>;
  distrito: Nullable<string>;
  referencia: Nullable<string>;
  telefono: Nullable<string>;
  email: Nullable<string>;
  responsable_nombre: Nullable<string>;
  responsable_cargo: Nullable<string>;
  horario_atencion: Nullable<string>;
  latitud: Nullable<number>;
  longitud: Nullable<number>;
  activo: boolean;
  es_principal: boolean;
  fecha_creacion: string;
  fecha_actualizacion: string;
}

export interface CrearSedeRequest {
  nombre: string;
  slug: string;
  codigo_sede?: string;
  direccion: string;
  departamento?: string;
  provincia?: string;
  distrito?: string;
  referencia?: string;
  telefono?: string;
  email?: string;
  responsable_nombre?: string;
  responsable_cargo?: string;
  latitud?: number;
  longitud?: number;
}
